scripts/generate-preview-build-message.ts

```diff
@@ -0,0 +1,54 @@
+#!yarn ts-node
+
+import fs from 'fs';
+import path from 'path';
+import execa from 'execa';
+
+/**
+ * Write a preview build message to the path "preview-build-message.txt".
+ */
+async function main() {
+  const packageMap: Record<string, string> = {};
+
+  const { stdout } = await execa('yarn', [
+    'workspaces',
+    'list',
+    '--no-private',
+    '--json',
+  ]);
+  const packages = stdout.split('\n').map((line) => JSON.parse(line));
+  const packageManifestsPaths = packages.map(({ location }) =>
+    path.join(location, 'package.json'),
+  );
+  for (const manifestPath of packageManifestsPaths) {
+    const rawManifest = await fs.promises.readFile(manifestPath, {
+      encoding: 'utf8',
+    });
+    const { name, version } = JSON.parse(rawManifest);
+
+    packageMap[name] = version;
+  }
+
+  const previewBuildMessage = `
+Preview builds have been published. [See these instructions](https://github.com/MetaMask/controllers/blob/main/docs/contributing.md#using-controllers-in-other-projects-during-developmenttesting) for more information about preview builds.
+
+<details>
+
+<summary>Expand for full list of packages and versions:</summary>
```

Comment:
Nit: I think it makes more sense to have a period here. Especially when collapsed, the `:` doesn't make sense.

```suggestion
<summary>Expand for full list of packages and versions.</summary>
```